components/support/nimbus-fml/src/backends/kotlin/templates/ObjectTemplate.kt

```diff
@@ -36,16 +36,19 @@ specify all values needed for the  feature #}
     {%- let prop_kt = p.name()|var_name %}
     {{ p.doc()|comment("    ") }}
     val {{ prop_kt }}: {{ p.typ()|type_label }} by lazy {
-        {%- let t = p.typ() %}
-        {%- let overrides = t|get_value("_variables?", p.name()) %}
         {%- let defaults = format!("_defaults.{}", prop_kt) %}
-        {{ t|with_fallback(overrides, defaults) }}
+        {{ p.typ()|property(p.name(), "_variables", defaults)}}
     }
     {%- endfor %}
 
+    internal fun mergeWith(defaults: {{class_name}}): {{class_name}} =
+        {{class_name}}(_variables = this._variables, _defaults = defaults._defaults)
+
     companion object {
-        fun create(variables: Variables): {{class_name}}? {
-            return {{class_name}}(_variables = variables)
-        }
+        internal fun create(variables: Variables): {{class_name}}? =
+            {{class_name}}(_variables = variables)
+
+        internal fun mergeWith(overrides: {{class_name}}, defaults: {{class_name}}): {{class_name}} =
+            overrides.mergeWith(defaults)
```

Comment:
Generating these methods makes such a difference to the recursive mapping and merging functions.

Reply:
🤩